### PR TITLE
Convert transferAccount reducer and sagas to typescript

### PR DESCRIPTION
**Pull request by gtaschuk**
**Describe the Pull Request**

Converts `transferAccount` reducers and sagas to typescript

#290 should go in first, and then I'll need to rebase this before it can be merged since there is some overlap in the sagas

### CODE REVIEW

**Comment by tristanhcole** [comment]
Is this ready for review @gtaschuk ?

**Comment by gtaschuk** [comment]
@tristanhcole yep

File: app/client/sagas/creditTransferSagas.ts [hunk 1]
```diff
@@ -49,7 +46,10 @@ function* updateStateFromCreditTransfer(result: CreditLoadApiResult) {
 
   const transfer_accounts = normalizedData.entities.transfer_accounts;
   if (transfer_accounts) {
-    yield put({ type: DEEP_UPDATE_TRANSFER_ACCOUNTS, transfer_accounts });
+    yield put({
+      type: TransferAccountActionTypes.DEEP_UPDATE_TRANSFER_ACCOUNTS,
+      transfer_accounts
+    });
```
Comment (tristanhcole) [inline app/client/sagas/creditTransferSagas.ts:52]: I think this should be an action... `TransferAccountAction.deepUpdateTransferAccounts()`

File: app/client/sagas/creditTransferSagas.ts [hunk 2]
```diff
@@ -62,7 +62,8 @@ function* updateStateFromCreditTransfer(result: CreditLoadApiResult) {
     // we need to add the newly created credit_transfer id
     // to the associated transfer_account object credit_transfer array
     yield put({
-      type: UPDATE_TRANSFER_ACCOUNTS_CREDIT_TRANSFERS,
+      type:
+        TransferAccountActionTypes.UPDATE_TRANSFER_ACCOUNTS_CREDIT_TRANSFERS,
       credit_transfer_list
     });
```
Comment (tristanhcole) [inline app/client/sagas/creditTransferSagas.ts:68]: Same here, should use an action `TransferAccountAction.updateTransferAccountsCreditTransfers`

File: app/client/components/pages/transferAccountListPage.jsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { LightTheme } from "../theme.js";
 import TransferAccountListWithFilterWrapper from "../transferAccount/transferAccountListWithFilterWrapper.jsx";
 import UploadButton from "../uploader/uploadButton.jsx";
 
-import { loadTransferAccounts } from "../../reducers/transferAccountReducer";
+import { LoadTransferAccountAction } from "../../reducers/transferAccount/reducers";
```
Comment (tristanhcole) [inline app/client/components/pages/transferAccountListPage.jsx:17]: This throws an error. Should be `"../../reducers/transferAccount/actions";`

File: app/client/reducers/creditTransfer/types.ts [hunk 4]
```diff
@@ -51,6 +50,18 @@ export interface CreditTransfer {
   uuid?: null | string;
 }
 
+export interface DisbursementCreditTransfer extends BaseCreditTransfer {
+  recipient_transfer_account: TransferAccount;
+}
+
+export interface ReclamationCreditTransfer extends BaseCreditTransfer {
+  sender_transfer_account: TransferAccount;
+}
```
Comment (tristanhcole) [inline app/client/reducers/creditTransfer/types.ts:59]: Nice

File: app/client/reducers/transferAccount/actions.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  createAction(
+    LoadTransferAccountActionTypes.LOAD_TRANSFER_ACCOUNTS_REQUEST,
+    payload
+  );
+
+export const loadTransferAccountsSuccess = (lastQueried: Date) =>
+  createAction(
+    LoadTransferAccountActionTypes.LOAD_TRANSFER_ACCOUNTS_SUCCESS,
+    lastQueried
+  );
```
Comment (tristanhcole) [inline app/client/reducers/transferAccount/actions.ts:66]: I think these payloads should be more explicit to take full advantage of Typescripts benefits. Console log the payloads and server response to see the data structure. See CreditTransfer actions for examples:

https://github.com/teamsempo/SempoBlockchain/blob/bb62b22656b8423e537a866464bedcd488b5f3c2/app/client/reducers/creditTransfer/actions.ts#L34-L47

File: app/client/reducers/transferAccount/types.ts [hunk 6]
```diff
@@ -16,3 +41,22 @@ export interface MultipleTransferAccountData {
 export type TransferAccountData =
   | SingularTransferAccountData
   | MultipleTransferAccountData;
+
+export interface TransferAccountLoadApiResult {
+  type: typeof LoadTransferAccountActionTypes.LOAD_TRANSFER_ACCOUNTS_REQUEST;
+  payload: any;
```
Comment (tristanhcole) [inline app/client/reducers/transferAccount/types.ts:47]: I think payloads should be explicitly typed

**Comment by tristanhcole** [comment]
Additionally, would be good to convert transferAccountAPI to typescript while we are at it.

**Pull request merged**